programs/pluto/ikev2_ts.c

```diff
@@ -125,13 +137,58 @@ static stf_status ikev2_emit_ts(pb_stream *outpbs,
 	{
 		struct ikev2_ts its = {
 			.isat_critical = ISAKMP_PAYLOAD_NONCRITICAL,
-			.isat_num = 1,
+			/*
+			 * If there is a security label in the Traffic Selector,
+			 * then we must send a TS_SECLABEL substructure as part of the
+			 * Traffic Selector (TS) Payload.
+			 * That means the TS Payload contains two TS substructures:
+			 *  - One for the address/port range
+			 *  - One for the TS_SECLABEL
+			 */
+			.isat_num = ts->sec_ctx ? 2 : 1,
 		};
 
 		if (!out_struct(&its, ts_desc, outpbs, &ts_pbs))
 			return STF_INTERNAL_ERROR;
 	}
 
+	/*
+	 * Output a TS_SECLABEL substructure as part of the TS Payload if a
+	 * security label exists.
+	 */
+	if (ts->sec_ctx) {
+		/* Sanity check: +1 is for the NUL-termination character. */
```

Comment:
NULL

Reply:
I meant it as the NUL character: https://en.wikipedia.org/wiki/Null_character